frontend/src/metabase/timelines/actions.ts

```diff
@@ -32,6 +32,28 @@ export const createTimelineWithEvent = createThunkAction(
   },
 );
 
+export const UPDATE_TIMELINE = "metabase/timelines/UPDATE_TIMELINE";
+export const updateTimeline = createThunkAction(
+  UPDATE_TIMELINE,
+  (timeline: Timeline, collection: Collection) => {
+    return async (dispatch: Dispatch) => {
+      await dispatch(Timelines.actions.update(timeline));
+      dispatch(push(Urls.timelineInCollection(timeline, collection)));
+    };
+  },
+);
+
+export const ARCHIVE_TIMELINE = "metabase/timelines/ARCHIVE_TIMELINE";
+export const archiveTimeline = createThunkAction(
+  ARCHIVE_TIMELINE,
+  (timeline: Timeline, collection: Collection) => {
+    return async (dispatch: Dispatch) => {
+      await dispatch(Timelines.actions.setArchived(timeline, true));
+      dispatch(push(Urls.timelinesInCollection(collection)));
+    };
+  },
+);
+
```

Comment:
Can we manage navigation in a different place? We'll need to create, update and archive events from the QB sidebar too and atm it will always navigate a user to the collections page

Reply:
I was thinking about that and my current thoughts are:
- Connect these modals to different actions for the QB
- Differentiate based on the current route

Do you have other ideas how to do that?

Reply:
WDYT about moving navigation logic to the container component? E.g. for `TimelineModal`:

```jsx
const mapDispatchToProps = dispatch => ({
   onArchive: async event => {
      await dispatch(archiveEvent(event))
      dispatch(push(Urls.whereeverWeNeed()))
   },
})
```

Reply:
@kulyk Fixed, please have a look